gpflow/training/monitor.py

```diff
@@ -621,6 +640,90 @@ def run(self, context: MonitorContext, *args, **kwargs) -> None:
                          global_step=context.global_step_tensor)
 
 
+class LogdirWriter(tf.summary.FileWriter):
+    """
+    This is a wrapper around the TensorFlow summary.EventWriter that provides a workaround for
+    a bug currently present in this module. The EventWriter can only open the file in exclusive
+    mode, however when multiple instances of the writer attempt to access the same file no error
+    is raised.
+
+    This class prevents user from opening multiple writers with the same location (event file
+    directory and file name suffix). It keeps a global set of used locations adding a new location
+    there when a writer is created or reopened. It removes the location from the global set when
+    the writer is closed or garbage collected.
+
+    Once the bug in TensorFlow is fixed this class can be removed or reduced to trivial:
+    class LogdirWriter(tf.summary.FileWriter):
+        pass
+    """
+
+    _used_locations = set()     # type: Set[Tuple[str, Optional[str]]
+
+    def __init__(self, logdir: str, graph: Optional[tf.Graph]=None, max_queue: int=10,
+                 flush_secs: float=120, filename_suffix: Optional[str]=None):
+        """
+        A thin wrapper around the summary.FileWriter __init__. It remembers the location (a tuple
+        of event file directory and file name suffix) and attempts to lock it. If the location
+        is already locked by another writer an error will be raised.
+
+        :param logdir: Directory where event file will be written.
+        :param graph: A `Graph` object
+        :param max_queue: Size of the queue for pending events and summaries.
+        :param flush_secs: How often, in seconds, to flush the added summaries and events to disk.
+        :param filename_suffix: Optional suffix of the event file's name.
+        """
+
+        self._location = (str(PurePath(logdir)), filename_suffix)
+        self._is_locked = False
```

Comment:
there is no need in extra `self._is_locked` variable, you can make a function which checks that location is present in locations set.

Reply:
Yes, but why is it better?

Reply:
no need in variables which actually do nothing extra on top of `in` operation for set

Reply:
I agree with Artem here, if we would rename `used_locations` to `locked_locations` we achieve the same functionality with less additional logic.

Reply:
My comment "Yes, but why is it better?" was misleading. The variable is needed. But I think we are now in agreement about this. I changed the names as per the suggestions.